### PR TITLE
Index Set Page Performance Improvements

### PR DESCRIPTION
**Pull request by grotlue**
Closes: https://github.com/Graylog2/graylog2-server/issues/12498

## Description

- Reduced amount of calls to the database
- Only get stats for relevant indices requested in the pagination request
- Added paginated api call for searching based on title

### UI:
- Added search field for paginated search
- Added button to toggle stats fetching on/off (disabled by default)

Screenshot 2023-04-24 at 16 02 49
Monosnap screencast 2023-04-24 16-03-56

## Motivation and Context

When a large number of indexes are present (or a few very large indexes are present), the Indices and Index Sets page and the details page for each index seem to be slow
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by mpfz0r — APPROVED** [review]
LGTM :+1:  
We won't be merging this into 5.1, right?

**Comment by kodjo-anipah** [comment]
> LGTM 👍 We won't be merging this into 5.1, right?

Yeah lets wait after 5.1 is done

File: graylog2-web-interface/src/components/indices/IndexSetsComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        <GlobalStatsCol md={3}>
+          <GlobalStats><strong>Stats for all indices:</strong> {statsEnabled
+            ? formatStatsString(globalIndexSetStats)
+            : <StatsInfoText>{statsDisabledText}</StatsInfoText>}
+          </GlobalStats>
+          <Button onClick={onToggleStats}>{statsEnabled ? 'Disable stats' : 'Enable stats'}</Button>
+        </GlobalStatsCol>
+        <Col md={3}>
+          <SearchForm onSearch={onSearch}
+                      label="Search"
```
Comment (gally47) [inline graylog2-web-interface/src/components/indices/IndexSetsComponent.tsx:219]: I think we should keep the search field UI consistent in all the views by removing the `label` and putting the field on the `left side`. For the stats info we can either put it below the search field or in the right side.

**Review by gally47 — CHANGES_REQUESTED** [review]
Frontend looking good, just one suggestion.

**Review by gally47 — APPROVED** [review]
LGTM!

**Pull request merged**